packages/cli/src/cli.ts

```diff
@@ -47,22 +50,76 @@ export function cli({
                         description:
                             'List of tables to be included in the generation',
                     })
-                    .array('tables');
+                    .array('tables')
+                    .config('configFile', function (configPath) {
+                        const data = JSON.parse(
+                            fs.readFileSync(configPath, 'utf-8'),
+                        );
+
+                        if (!validateConfigFile(data)) {
+                            throw new Error(
+                                [
+                                    `Config file at path '${configPath}' does not match the schema:`,
+                                    ...(validateConfigFile.errors ?? []),
+                                ].join('\n'),
+                            );
+                        }
+                        return data;
+                    })
+                    .parserConfiguration({
+                        'dot-notation': false,
+                        'strip-aliased': true,
+                    });
             },
             async (argv) => {
                 console.warn('Connecting to database schema...');
 
-                const { schema } = await generateSchema({
-                    connectionString: argv.connectionString,
-                    out: argv.out,
-                    defaultSchema: argv.defaultSchema,
-                    schemas: argv.schemas,
-                    tables: argv.tables,
-                });
-                const tables = Object.keys(schema.properties ?? {});
-                console.warn(`Generated schema with ${tables.length} table(s)`);
-                for (const table of tables) {
-                    console.warn(`- ${table}`);
+                // We do this so that we lose the types from yargs but we gain
+                // complete type-safety after we pass the data through the validator
+                const data = { ...argv, connectionString: undefined } as any;
```

Comment:
Don't use `as any`. For cases like these use `as unknown`. 

`as any` is dangerous, `as unknown` is safe.
`as any` allows any compiler errors because it allows any type.
`as unknown` will force you to check for the type, so it's totally safe to use it.